crates/sui-node/src/lib.rs

```diff
@@ -388,11 +417,40 @@ impl SuiNode {
             .ok_or_else(|| anyhow::anyhow!("Transaction Orchestrator is not enabled in this node."))
     }
 
-    //TODO watch/wait on all the components
-    pub async fn wait(self) -> Result<()> {
-        self.grpc_server.await??;
-
-        Ok(())
+    pub async fn monitor_reconfiguration(mut self) -> Result<()> {
+        loop {
+            let next_epoch = self
+                .reconfig_channel
```

Comment:
Do we expect this channel to be pre-populated with a message - meaning do we expect the code on the `monitor_reconfiguration` to run on bootstrap as well to sort of initialise values - or this will happen only in new epoch cases?

Reply:
It should only happen when in new epoch cases.